### PR TITLE
use tidb region cache to manage region and stores

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

Currently the kv client simply calls pd `ScanRegions` when the event feed rpc meets some error, such as `NotLeader`, `gPRC` failed etc. However the `ScanRegions` API not always returns the realtime available stores and leader information. So we need a better way to handle region and store information.

### What is changed and how it works?
TiDB region cache has provided a convenient to handle region or store error, we can reuse this lib.

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: cdc/kv/client.go [hunk 1]
```diff
@@ -226,24 +194,30 @@ func (c *CDCClient) partialRegionFeed(
...
-		if regionInfo.meta == nil {
-			regionInfo.meta, _, err = c.pd.GetRegion(ctx, regionInfo.span.Start)
-		}
-
+		bo := tikv.NewBackoffer(ctx, tikvRequestMaxBackoff)
+		rpcCtx, err := c.regionCache.GetTiKVRPCContext(bo, regionInfo.verID, tidbkv.ReplicaReadLeader, 0)
 		if err != nil {
-			log.Warn("get meta failed", zap.Error(err))
-			return errors.Trace(err)
+			return backoff.Permanent(errors.Trace(err))
```
[the remark this answers is not in the document]
Comment (amyangfei) [reply cdc/kv/client.go]: `GetTiKVRPCContext` will re-fetch a region from region cache in every `singleEventFeed` retry.
Besides the `GetTiKVRPCContext` itself has builtin retry, which is controlled by `tikv.Backoffer`

File: cdc/kv/client.go [hunk 2]
```diff
@@ -226,24 +194,30 @@ func (c *CDCClient) partialRegionFeed(
...
 		}
+		if rpcCtx == nil {
+			select {
+			case <-ctx.Done():
```
[the remark this answers is not in the document]
Comment (amyangfei) [reply cdc/kv/client.go]: If the passed-in `ctx` is canceled, the `rpcCtx` and `err` returned from `GetTiKVRPCContext` could be both nil, we can break the logic as soon as possible.

**Comment by leoppro** [comment]
LGTM

**Pull request merged**